packages/hdwallet-provider/src/index.ts

```diff
@@ -215,18 +217,97 @@ class HDWalletProvider {
     });
   }
 
-  public send(
+  private initialize(): Promise<void> {
+    return new Promise((resolve, reject) => {
+      this.engine.sendAsync({
+        jsonrpc: '2.0',
+        id: Date.now(),
+        method: 'eth_chainId',
+        params: []
+    }, (error: any, response: JSONRPCResponsePayload) => {
+        if (error) {
+          reject(error);
+        }
+        if (response && response.result) {
```

Comment:
The node may respond with `response.error`, in which case we should handle that by rejecting.

Reply:
So I got rid of this check for `response` and `response.result` since the spec implies one or the other will be there. I also created a utility type to account for the `error` field and reject if it is there.